media/components/sdfv/properties/properties.js

```diff
@@ -0,0 +1,267 @@
+        this.element = element;
+        this.target = target;
+        this.key = key;
+        this.subkey = subkey;
+        this.datatype = datatype;
+    }
+
+    write_back(value) {
+        if (this.subkey !== undefined) {
+            if (this.datatype === 'Range' ||
+                this.datatype === 'SubsetProperty') {
+                if (this.target[this.key])
+                    this.target[this.key][this.subkey] = value;
+                else
+                    this.target[this.key] = {
+                        type: 'Range',
+                        ranges: value,
+                    };
+            } else {
+                this.target[this.key][this.subkey] = value;
+            }
+        } else {
+            this.target[this.key] = value;
+        }
+
+        if (this.key === 'label') {
+            // If the label was changed, we want to update the renderer graph
+            // element label as well.
+            if (this.element.data) {
+                if (this.element.data.node) {
+                    this.element.data.node.label = value;
+
+                    if (this.element instanceof ScopeNode) {
+                        // In scope nodes the range is attached.
+                        if (this.element instanceof EntryNode) {
+                            let exit_elem = find_graph_element_by_uuid(
+                                renderer.graph,
+                                this.element.sdfg.sdfg_list_id + '/' +
+                                this.element.parent_id + '/' +
+                                this.element.data.node.scope_exit + '/-1'
+                            );
+                            if (exit_elem) {
+                                this.element.data.node.label =
+                                    compute_scope_label(this.element);
+                                exit_elem.element.data.node.label =
+                                    this.element.data.node.label;
+                            }
+                        } else if (this.element instanceof ExitNode) {
+                            let entry_elem = find_graph_element_by_uuid(
+                                renderer.graph,
+                                this.element.sdfg.sdfg_list_id + '/' +
+                                this.element.parent_id + '/' +
+                                this.element.data.node.scope_entry + '/-1'
+                            );
+                            if (entry_elem) {
+                                this.element.data.node.label =
+                                    compute_scope_label(entry_elem.element);
+                                entry_elem.element.data.node.label =
+                                    this.element.data.node.label;
+                            }
+                        }
+                    }
+                }
+            }
+        }
+    }
+
+    get_value() {}
+
+    update() {}
+
+}
+
+class KeyProperty {
```

Comment:
Does not extend Property?

Reply:
originally by design because although the naming is simlar, the two behaved pretty differently. It has changed so much at this point that it technically could. Needs some adaptation for it though